src/validation/schema.js

```diff
@@ -389,6 +389,282 @@ const validateNoImportDirective = def =>
+    ? validateFullTextDirectiveArgumentAlgorithm(def, directive, algorithm)
+    : List([
+        immutable.fromJS({
+          loc: directive.name.loc,
+          entity: def.name.value,
+          message: `@fulltext argument 'algorithm' must be specified`,
+        }),
+      ])
+}
+
+const validateFullTextDirectiveArgumentAlgorithm = (def, directive, argument) => {
+  if (argument.value.kind != 'EnumValue') {
+    return List().push(
+      immutable.fromJS({
+        loc: directive.name.loc,
+        entity: def.name.value,
+        message: `@fulltext argument 'algorithm' must be an enum`,
+      }),
+    )
+  } else if (!['RANKED', 'PROXIMITY_RANK'].includes(argument.value.value)) {
+    return List().push(
+      immutable.fromJS({
+        loc: directive.name.loc,
+        entity: def.name.value,
+        message: `@fulltext 'algorithm' value, '${argument.value.value}', is not a variant of the _FullTextAlgorithm enum`,
+      }),
+    )
+  } else {
+    return List([])
+  }
+}
+
+const validateFullTextDirectiveInclude = (def, directive) => {
+  let include = directive.arguments.find(argument => argument.name.value == 'include')
+  if (include) {
+    if (include.value.kind != 'ListValue') {
+      return List().push(
+        immutable.fromJS({
+          loc: directive.name.loc,
+          entity: def.name.value,
+          message: `@fulltext argument 'include' must be a list`,
+        }),
+      )
+    }
+    return include.value.values.reduce(
+      (errors, type) =>
+        errors.concat(validateFullTextDirectiveArgumentInclude(def, directive, type)),
+      List(),
+    )
+  } else {
+    return List([
+      immutable.fromJS({
+        loc: directive.name.loc,
+        entity: def.name.value,
+        message: `@fulltext argument 'include' must be specified`,
+      }),
+    ])
+  }
+}
+
+const validateFullTextDirectiveArgumentInclude = (def, directive, argument) => {
+  if (argument.kind != 'ObjectValue') {
+    return List().push(
+      immutable.fromJS({
+        loc: directive.name.loc,
+        entity: def.name.value,
+        message: `@fulltext argument 'include' must be an object`,
+      }),
+    )
+  }
+  if (argument.fields.length != 2) {
+    return List([
+      immutable.fromJS({
+        loc: directive.name.loc,
+        entity: def.name.value,
+        message: `fulltext include argument must have two fields, 'entity' and 'fields'}`,
+      }),
+    ])
+  }
+  return argument.fields.reduce(
+    (errors, field) =>
+      errors.concat(
+        validateFulltextDirectiveArgumentIncludeFields(def, directive, field),
+      ),
+    List([]),
+  )
+}
+
+const validateFulltextDirectiveArgumentIncludeFields = (def, directive, field) => {
+  if (!['entity', 'fields'].includes(field.name.value)) {
+    return List().push(
+      immutable.fromJS({
+        loc: directive.name.loc,
+        entity: def.name.value,
+        message: `@fulltext field '${field.name.value}' invalid, may only be one of: [entity, fields]`,
```

Comment:
`field '${field.name.value}'` -> `argument 'include > ${field.name.value}'` perhaps, to make the hierarchy clear in the error?